src/python/tritonfrontend/_api/_kservegrpc.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+from enum import IntEnum
+from typing import Union
+
+import tritonserver
+from pydantic import Field
+from pydantic.dataclasses import dataclass
+from tritonfrontend._c.tritonfrontend_bindings import TritonFrontendGrpc
+
+
+# Enum (mirroring C++ format)
+class Grpc_compression_level(IntEnum):
+    NONE = 0
+    LOW = 1
+    MED = 2
+    HIGH = 3
+    COUNT = 4
+
+
+class KServeGrpc:
+    Grpc_compression_level = (
+        Grpc_compression_level  # Include the enum as a class attribute
+    )
+
+    # triton::server::grpc::Options
+    @dataclass
+    class Options:
```

Comment:
Is grpc missing `restricted_protocols`?

Reply:
Not supported, will be addressed in a follow up PR.

Reply:
Please add inline comment referencing the JIRA ticket ("DLIS-1234", not the full link) for follow-up:
```
# DLIS-XXXX: Add restricted protocol support
```